lib/ansible/plugins/connection/network_cli.py

```diff
@@ -383,13 +384,21 @@ def receive(self, command=None, prompts=None, answer=None, newline=True, prompt_
         self._matched_cmd_prompt = None
         matched_prompt_window = window_count = 0
 
+        signal_handler = signal.getsignal(signal.SIGALRM)
+        if signal_handler == signal.SIG_DFL:
+            signal_handler = None
+
         while True:
             data = self._ssh_shell.recv(256)
 
             # when a channel stream is closed, received data will be empty
             if not data:
                 break
 
+            if signal_handler:
```

Comment:
Since `receive()` is invoked immediately after the connection to remote host is successful [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/connection/network_cli.py#L315) to fetch the initial prompt from remote host, for this particular scenario it is not required to reset the timer since if initial prompt doesn't match terminal regex as it might go into infinite loop. Hence will be safe to add a check to reset timer if `self._matched_prompt` value is also set.
```
if signal_handler and self._matched_prompt:
```

Reply:
@ganeshrn I've looked over the code you mentioned, however I am struggling to see how this could go into an infinite loop, can you please elaborate?

The timer is only extended when the device outputs data, so yes the time limit to see the first prompt will be extended slightly, but as long as the device is not printing output to the terminal that continues to fail to match a prompt, the timer will still expire.  
```
if not data:
   break
```
When there is no output the above prevents the timer from being extended..